### PR TITLE
[aos-clearhsm] Add script to clear hsm storage

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    file://aos-clearhsm.sh \
+"
+
+S = "${WORKDIR}"
+
+FILES:${PN} = " \
+    ${bindir} \
+"
+
+RDEPENDS:${PN} = "aos-setupdisk"
```
Comment (MykolaSolyanko) [inline meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb:16]: why is this dependency needed?

**Review by MykolaSolyanko — APPROVED** [review]

File: meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+DESCRIPTION = "Aos script to clear PKCS11 storage"
```
Comment (MykolaSolyanko) [inline meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb:1]: should we move this receipt to meta-aos? I mean this functionality be used in another target not only for VM?

**Review by MykolaSolyanko — APPROVED** [review]

File: meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb (hunk 1 above)
Comment (mykola-kobets-epam) [reply meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb]: it is not needed. forgot to remove while adapting aos-deprov.sh

File: meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb (hunk 2 above)
Comment (mykola-kobets-epam) [reply meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb:1]: I guess it will be difficult because aos-clearhsm.sh will be different.

Comment (al1img) [reply meta-aos-vm-common/recipes-aos/aos-clearhsm/aos-clearhsm.bb:1]: It is platform specific. Should be in meta-aos-vm.

File: meta-aos-vm-common/recipes-core/images/aos-image-vm.bb [hunk 3]
```diff
@@ -26,6 +26,7 @@ IMAGE_INSTALL:append = " \
 # AOS packages
 IMAGE_INSTALL:append = " \
     aos-deprov \
+    aos-clearhsm \
```
Comment (al1img) [inline meta-aos-vm-common/recipes-core/images/aos-image-vm.bb:29]: As agreed we should call the script before IAM in aos-iamanager-provisioning.service. So, it should be added as rdepend in the corresponding recipe.

**Review by mlohvynenko — APPROVED** [review]

File: meta-aos-vm-common/recipes-aos/aos-deprov/files/deprovision-async.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
Comment (al1img) [inline meta-aos-vm-common/recipes-aos/aos-deprov/files/deprovision-async.sh:1]: as discussed let's have one script with parameters for different scenarios.

**Review by al1img — APPROVED** [review]

**Pull request merged**